src/utilities/httpUtils.js

```diff
@@ -0,0 +1,69 @@
+import axios from 'axios';
+import urlConstants from '~/constant/urlConstants';
+import textConstants from '~/constant/textConstants';
+import * as localStorage from '~/storage/LocalStorage';
+
+const http = axios.create({
+  baseURL: urlConstants.apiBaseUrl,
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+
+http.interceptors.response.use(
+  response => {
+    return response;
+  },
+  error => {
+    const originalRequest = error.config;
+
+    if (
+      error.response.status === textConstants.UNAUTHORIZED_CODE &&
+      error.response.data.error &&
+      error.response.data.error.message === textConstants.ACCESS_TOKEN_EXPIRE
+    ) {
+      const refreshToken = localStorage.getRefreshToken();
+
+      return http.post(urlConstants.refreshTokenUrl, { authorization: `Bearer ${refreshToken}` }).then(({ data }) => {
+        const accessToken = data.accessToken;
+
+        localStorage.saveAccessToken(accessToken);
+        originalRequest.headers.Authorization = `Bearer ${accessToken}`;
+
+        return http(originalRequest);
+      });
+    } else if (
+      originalRequest.url !== urlConstants.googleLoginUrl &&
+      ((error.response.status === textConstants.UNAUTHORIZED_CODE &&
```

Comment:
@nikitashrestha this conditioning is too complicated. Would you like to explain what are you trying to accomplish and what conditions ou are considering?

Reply:
when the user is sending request except for login, and if in case the server responds with unauthorized / token not found / refresh token expired then user needs to be logged out